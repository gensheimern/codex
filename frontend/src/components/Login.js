import React from 'react';

class Login extends React.Component {

	constructor(props) {
		super(props);

		this.inputEmail = this.inputEmail.bind(this);
		this.inputPassword = this.inputPassword.bind(this);
		this.handleClick = this.handleClick.bind(this);

		this.state = {
			email: "",
			pw: ""
		}
	}

	inputEmail(e) {
		this.setState({
			email: e.target.value
		});
	}

	inputPassword(e) {
		this.setState({
			pw: e.target.value
		});
	}

	handleClick(e) {
		e.preventDefault();

		console.log("Mail: " + this.state.email);
		console.log("Password: " + this.state.pw);
	}

	render() {
		return(
			<div style={{
				textAlign: 'center',
				marginTop: '5%',
				border: '1px solid black'
			}}>
				<form>
					<label>E-Mail:</label>
					<input type="email" value={this.state.email} onChange={this.inputEmail} />
					<br />
					<label>Password:</label>
					<input type="password" value={this.state.pw} onChange={this.inputPassword} />
					<button onClick={this.handleClick}>Login</button>
				</form>
			</div>
		);
	}
}

export default Login;