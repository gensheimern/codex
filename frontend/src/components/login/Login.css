.centered {
	text-align: center;
}

.loginArea {
	width: 300px;
	font-size: 15pt;
	text-align: right;
	margin: auto;
	margin-top: 20%;
	padding: 1% 5%;
	border: 1px solid blue;
}