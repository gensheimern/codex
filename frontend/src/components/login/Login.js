import React, {
    Component
} from "react";
import PropTypes from 'prop-types';
import "./Login.css";
import config from '../../config';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import {Card,CardText} from 'material-ui/Card';
import Paper from 'material-ui/Paper';
import { Link } from 'react-router-dom';
import Logout from './Logout';

export default class Login extends Component {

    constructor(props) {
        super(props);

        this.state = {
            email: "",
            password: "",
            errorPrompt: "",
            open: false,
            errorText: "",
        };
    }

    handleOpen = () => {
        this.setState({open: true});
    };

    handleClose = () => {
        this.setState({open: false});
    };

    handleChange = name => e => {
		this.setState({
			[name]: e.target.value
		});
	}

    validateForm() {
        return this.state.email.length > 0 && this.state.password.length > 0;
    }


    handleSubmit = event => {
        event.preventDefault();

        fetch(config.apiPath + "/authenticate", {
            method: 'POST',
            body: JSON.stringify({
                email: this.state.email,
                password: this.state.password
            }),
            headers: {
                'Content-Type': 'application/json'
            }
        }).then((res) => {
            if(!res.ok) {
                this.setState({errorText:'Invalid Password or Email'})
                throw new Error("Invalid Password");
            } else if(res.status !== 200) {
                throw new Error("Forbidden");
            }
            return res.json();
        })
        .then((res) => {

            if(typeof (Storage) !== "undefined") {
                localStorage.setItem("apiToken", res.token);
            } else {
                // TODO Code without local storage
            }

            this.props.history.push("/feed");
        }).catch((err) => {
            
        });
    }

	render() {
        const styles ={

            paperStyle:{
				height: 62,
				width: 62,
				borderRadius: 7,
				display: 'block',
				marginLeft: 'auto',
				marginRight: 'auto',
            },

            textField:{
				width: '80%',
				marginLeft: '10%',
				marginRight: '10%',
            },

            loginButton:{
				heigt: 40.57,
				width: '30%',
				borderRadius: 3,
				backgroundColor: "#1EA185",
				display: 'block',
				marginLeft: 'auto',
				marginRight: 'auto',
			},
        }

        return(
            <div className = "loginBg">
            <form className ="login" >
                <div>
                    <Paper style={styles.paperStyle} zDepth= {1}/>
                </div>
                <center>
                    <h3 className = "h3header">Lunchplannner</h3>
                </center>
                <Card className = "loginCard">
                <CardText>
                <h2 className = "h2header">Login</h2>
                <TextField
                    id="email"
					label="E-Mail"
					value={this.state.email}
					onChange={this.handleChange('email')}
					floatingLabelText="E-Mail"
					style= {styles.textField}

                />
                <TextField
					id="password"
					label="Password"
					type="password"
					value={this.state.password}
					onChange={this.handleChange('password')}
					floatingLabelText="Password"
					style= {styles.textField}
				/>

                <div className="errorText"> {this.state.errorText} </div>

                <Link to = "">Forgot password?</Link>
                <br/>
                <br/>
                <RaisedButton
					backgroundColor="#b9b9b9"
					type="submit"
					label="Login"
					primary={true}
					disabled={!this.validateForm()}
					style = {styles.loginButton}
                    onClick={this.handleSubmit}
				/>
                <br/>
                <p>
                    Don't have an account yet?&nbsp;
					<Link to = "/signup">Create an account</Link>
				</p>
                </CardText>
                </Card>
            </form>
            {this.props.logout ? <Logout /> : null}
            </div>
            );

	}
}

Login.propTypes = {
    logout: PropTypes.bool,
};

Login.defaultProps = {
    logout: false,
};
