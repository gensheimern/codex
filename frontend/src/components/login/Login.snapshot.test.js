import React from 'react';
import renderer from'react-test-renderer';

import Login from './Login';

describe('Login Snapshot', () => {
    test('renders', () =>{
        const component = renderer.create(
            <Login/>
        );
        let tree= component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});