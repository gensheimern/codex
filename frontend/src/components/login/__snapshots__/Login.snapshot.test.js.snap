// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Snapshot renders 1`] = `
<div
  className="loginBg"
>
  <form
    className="login"
  >
    <div>
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 7,
            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                   0 1px 4px rgba(0, 0, 0, 0.12)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "block",
            "fontFamily": "Roboto, sans-serif",
            "height": 62,
            "marginLeft": "auto",
            "marginRight": "auto",
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": 62,
          }
        }
      />
    </div>
    <center>
      <h3
        className="h3header"
      >
        Lunchplannner
      </h3>
    </center>
    <div
      className="loginCard"
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "zIndex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": 8,
          }
        }
      >
        <div
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontSize": 14,
              "muiPrepared": true,
              "padding": 16,
            }
          }
        >
          <h2
            className="h2header"
          >
            Login
          </h2>
          <div
            className={undefined}
            style={
              Object {
                "backgroundColor": "transparent",
                "cursor": "auto",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 16,
                "height": 72,
                "lineHeight": "24px",
                "marginLeft": "10%",
                "marginRight": "10%",
                "muiPrepared": true,
                "position": "relative",
                "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "80%",
              }
            }
          >
            <label
              className={undefined}
              htmlFor="email"
              onClick={undefined}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.3)",
                  "lineHeight": "22px",
                  "muiPrepared": true,
                  "pointerEvents": "none",
                  "position": "absolute",
                  "top": 38,
                  "transform": "scale(1) translate(0, 0)",
                  "transformOrigin": "left top",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "zIndex": 1,
                }
              }
            >
              E-Mail
            </label>
            <input
              disabled={false}
              id="email"
              label="E-Mail"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "WebkitOpacity": 1,
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "appearance": "textfield",
                  "backgroundColor": "rgba(0,0,0,0)",
                  "border": "none",
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "cursor": "inherit",
                  "font": "inherit",
                  "height": "100%",
                  "marginTop": 14,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "width": "100%",
                }
              }
              type="text"
              value=""
            />
            <div>
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              />
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 2,
                    "borderColor": "#00bcd4",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "transform": "scaleX(0)",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
          <div
            className={undefined}
            style={
              Object {
                "backgroundColor": "transparent",
                "cursor": "auto",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 16,
                "height": 72,
                "lineHeight": "24px",
                "marginLeft": "10%",
                "marginRight": "10%",
                "muiPrepared": true,
                "position": "relative",
                "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "80%",
              }
            }
          >
            <label
              className={undefined}
              htmlFor="password"
              onClick={undefined}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.3)",
                  "lineHeight": "22px",
                  "muiPrepared": true,
                  "pointerEvents": "none",
                  "position": "absolute",
                  "top": 38,
                  "transform": "scale(1) translate(0, 0)",
                  "transformOrigin": "left top",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "zIndex": 1,
                }
              }
            >
              Password
            </label>
            <input
              disabled={false}
              id="password"
              label="Password"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "WebkitOpacity": 1,
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "appearance": "textfield",
                  "backgroundColor": "rgba(0,0,0,0)",
                  "border": "none",
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "cursor": "inherit",
                  "font": "inherit",
                  "height": "100%",
                  "marginTop": 14,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "width": "100%",
                }
              }
              type="password"
              value=""
            />
            <div>
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              />
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": 2,
                    "borderColor": "#00bcd4",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "transform": "scaleX(0)",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
          <div
            className="errorText"
          >
             
            
             
          </div>
          <a
            href="/"
            onClick={[Function]}
          >
            Forgot password?
          </a>
          <br />
          <br />
          <div
            className={undefined}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "backgroundColor": "#1EA185",
                "borderRadius": 3,
                "boxShadow": undefined,
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "block",
                "fontFamily": "Roboto, sans-serif",
                "heigt": 40.57,
                "marginLeft": "auto",
                "marginRight": "auto",
                "minWidth": 88,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "30%",
              }
            }
          >
            <button
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "backgroundColor": "rgb(229, 229, 229)",
                  "border": 10,
                  "borderRadius": 2,
                  "boxSizing": "border-box",
                  "cursor": "default",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "height": 36,
                  "lineHeight": "36px",
                  "margin": 0,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "textAlign": "center",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "verticalAlign": null,
                  "width": "100%",
                }
              }
              tabIndex={-1}
              type="submit"
            >
              <div
                style={
                  Object {
                    "backgroundColor": false,
                    "borderRadius": 2,
                    "height": 36,
                    "muiPrepared": true,
                    "top": 0,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.3)",
                      "fontSize": 14,
                      "fontWeight": 500,
                      "letterSpacing": 0,
                      "margin": 0,
                      "muiPrepared": true,
                      "opacity": 1,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                      "position": "relative",
                      "textTransform": "uppercase",
                      "userSelect": "none",
                    }
                  }
                >
                  Login
                </span>
              </div>
            </button>
          </div>
          <br />
          <p>
            Don't have an account yet? 
            <a
              href="/signup"
              onClick={[Function]}
            >
              Create an account
            </a>
          </p>
        </div>
      </div>
    </div>
  </form>
</div>
`;
